benchmarking/platforms/ios/idb.py

```diff
@@ -41,8 +42,17 @@ def pull(self, src, tgt):
         return self.run("--download", src, "--to", tgt)
 
     def reboot(self):
-        # No reboot functionality
-        pass
+        # use idevicediagnostics to reboot device if exists
+        try:
+            super(IDB, self).run("idevicepair",
+                                 "-u", self.device, "pair")
+            super(IDB, self).run("idevicediagnostics",
+                                 "-u", self.device, "restart")
+            return True
+        except Exception:
+            getLogger().error("Rebooting failure...")
+            getLogger().error("Unknown exception {}".format(sys.exc_info()[0]))
```

Comment:
What's the difference if we just use the caught exception?

Reply:
We don't want the script to terminate in that case. Or at least some part of the code may be missed until the next try...except... clause.